lib/puppet/application/master.rb

```diff
@@ -196,6 +196,7 @@ def main
     if options[:rack]
       start_rack_master
     else
+      Puppet.deprecation_warning("The WEBrick Puppet master server is deprecated and will be removed in a future release.")
```

Comment:
Maybe mention that people should use puppetserver instead? Or maybe we need a docs page and perma-link about the webrick/rack deprecations? /cc @nfagerlund @ahpook